ports/nrf/modules/machine/uart.c

```diff
@@ -323,27 +338,36 @@ STATIC MP_DEFINE_CONST_DICT(machine_hard_uart_locals_dict, machine_hard_uart_loc
 STATIC mp_uint_t machine_hard_uart_read(mp_obj_t self_in, void *buf_in, mp_uint_t size, int *errcode) {
     const machine_hard_uart_obj_t *self = self_in;
     byte *buf = buf_in;
+    uint32_t t = self->timeout + mp_hal_ticks_ms();
 
     // read the data
     for (size_t i = 0; i < size; i++) {
         while (!uart_rx_any(self)) {
+            if (mp_hal_ticks_ms() > t) {  // timed out
```

Comment:
This won't work: eg `t` could be 0 after a wraparound and then this will time out immediately.  Instead it needs to be `mp_hal_ticks_ms() - t0 >= timeout`

Reply:
Done.